src/main/java/sage/Sage.java

```diff
@@ -0,0 +1,221 @@
+package sage;
+
+import sage.command.*;
+import sage.storage.Storage;
+import sage.task.*;
+import sage.parser.Parser;
+import sage.exception.SageException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * The main class for the Sage task management application
+ * This class initializes the application components, manages user interactions,
+ * and controls the overall flow of the application
+ */
+public class Sage {
+    public static final String DIRECTORY_PATH = "data/";
+    public static final String FILE_PATH = DIRECTORY_PATH + "/sage.txt";
+    private Storage storage;
+    private Ui ui;
+    private TaskList tasks;
+    private boolean isExit;
+    private CommandType commandType;
+
+    /**
+     * Constructs a Sage instance with the specified file path
+     * Initializes the UI, storage, parser, and task list.
+     * Attempts to load tasks from the file; if unsuccessful, initializes with an empty task list
+     *
+     */
+    public Sage() {
+        this.ui = new Ui();
+        this.storage = new Storage(FILE_PATH);
+        this.isExit = false;
+        try {
+            this.tasks = new TaskList(this.storage.load());
+        } catch (SageException e) {
+            this.ui.showErrorMessage(e);
+        }
+    }
+
+    public String getResponse(String input) {
+        try {
+            commandType = Parser.parseCommand(input);
+            switch (commandType) {
+                case BYE :
+                    this.exit();
+                    return this.ui.showGoodbyeMessage();
+                case LIST:
+                    return this.ui.showTaskList(tasks);
+                case MARK:
+                    return markCommand(Parser.parseArgs(CommandType.MARK, input));
+                case UNMARK:
+                    return unmarkCommand(Parser.parseArgs(CommandType.UNMARK, input));
+                case DELETE:
+                    return deleteCommand(Parser.parseArgs(CommandType.DELETE, input));
+                case FIND:
+                    return findCommand(Parser.parseArgs(CommandType.FIND, input));
+                case TODO:
+                    return toDoCommand(Parser.parseArgs(CommandType.TODO, input));
+                case DEADLINE:
+                    return deadlineCommand(Parser.parseArgs(CommandType.DEADLINE, input));
+                case EVENT:
+                    return eventCommand(Parser.parseArgs(CommandType.EVENT, input));
+                default:
+                    throw new SageException("Sorry, what do you mean? :p");
+            }
+        } catch (SageException e) {
+            return this.ui.showErrorMessage(e);
+        }
+    }
```

Comment:
Personally, I think that some of these might be classified under the Parser class rather than the main class, but that could just be something of personal preference that you could consider!